CentrED/UI/Windows/HistoryWindow.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System.Numerics;
+using CentrED.Client;
+using CentrED.Network;
+using ImGuiNET;
+using static CentrED.Application;
+
+namespace CentrED.UI.Windows;
+
+public class HistoryWindow : Window
+{
+    public override string Name => "History";
+    public override ImGuiWindowFlags WindowFlags => ImGuiWindowFlags.AlwaysAutoResize | ImGuiWindowFlags.NoResize;
+    
+    protected override void InternalDraw()
+    {
+        if (CEDClient.UndoStack.Count == 0)
+        {
+            ImGui.Text("History is empty.");
+            return;
+        }
+        
+        ImGui.Text($"Tasks in history: {CEDClient.UndoStack.Count}");
+        
+        // Create table
+        if (ImGui.BeginTable("UndoTable", 2, ImGuiTableFlags.Borders | ImGuiTableFlags.SizingFixedFit))
+        {
+            // Set column widths
+            ImGui.TableSetupColumn("Task", ImGuiTableColumnFlags.WidthFixed, 150);
+            ImGui.TableSetupColumn("Details", ImGuiTableColumnFlags.WidthFixed, 350);
+
+            // Headers
+            ImGui.TableHeadersRow();
+
+            var cnt = 0;
+            foreach (var command in CEDClient.UndoStack)
+            {
+                cnt++;
+                ImGui.TableNextRow();
+                
+                if (command.Length == 1)
+                {
+                    ImGui.TableNextColumn();
+                    GetHistory(command[0], out var task, out var details);
+                    ImGui.Text(task);
+                    ImGui.TableNextColumn();
+                    ImGui.Text(details);
+                }
+                else
+                {
+                    ImGui.TableNextColumn();
+                    ImGui.Text("Task collection");
+                    ImGui.TableNextColumn();
+                    
+                    foreach (var packet in command)
+                    {
+                        ImGui.TableNextRow();
+                        ImGui.TableNextColumn();
+                        GetHistory(packet, out var task, out var details);
+                        ImGui.Text($"\t {task}");
+                        ImGui.TableNextColumn();
+                        ImGui.Text(details);
+                    }
+                }
+
+                if (cnt >= 25)
```

Comment:
Why do we limit to only 25 entries? ImGUI is very good at providing a scrolling interface if stuff wouldn't fit inside the window and if this is due to performance we can use the imgui list clipper